open-metadata-implementation/access-services/data-platform/data-platform-server/src/main/java/org/odpi/openmetadata/accessservices/dataplatform/listeners/DataPlatformInTopicListener.java

```diff
@@ -49,26 +59,25 @@ public DataPlatformInTopicListener(OMEntityDao omEntityDao, OMRSAuditLog auditLo
     @Override
     public void processEvent(String eventAsString) {
 
-        NewViewEvent event = null;
+
         try {
-            event = OBJECT_MAPPER.readValue(eventAsString, NewViewEvent.class);
-        } catch (Exception e) {
-            DataPlatformErrorCode errorCode = DataPlatformErrorCode.PARSE_EVENT_EXCEPTION;
-            auditLog.logException("processEvent",
-                    errorCode.getErrorMessageId(),
-                    OMRSAuditLogRecordSeverity.EXCEPTION,
-                    errorCode.getFormattedErrorMessage(),
-                    "event {" + eventAsString + "}",
-                    errorCode.getSystemAction(),
-                    errorCode.getUserAction(),
-                    e);
-        }
-        if (event != null) {
-            try {
-                log.info("Started processing event in DataPlatform OMAS");
+            DataPlatformEventHeader dataPlatformEventHeader = OBJECT_MAPPER.readValue(eventAsString, DataPlatformEventHeader.class);
+
+            if (dataPlatformEventHeader.getEventType() == DataPlatformEventType.NEW_DEPLOYED_DB_SCHEMA_EVENT) {
```

Comment:
This if-else-if blocks looks like it will get to more conditions, it is worth to rethink of using a design pattern like strategy for future PR.

Reply:
Good suggestion. I will mark it as TODO